numpy 2.0.2 fails to build with Apple clang 17

```
  [numpy-2.0.2]   [spkg-install] In file included from ../numpy/_core/src/umath/string_ufuncs.cpp:20:
  [numpy-2.0.2]   [spkg-install] ../numpy/_core/src/umath/string_fastsearch.h:132:5: error: no type named 'ptrdiff_t' in namespace 'std'; did you mean simply 'ptrdiff_t'?
  [numpy-2.0.2]   [spkg-install]   132 |     std::ptrdiff_t
  [numpy-2.0.2]   [spkg-install]       |     ^~~~~~~~~~~~~~
  [numpy-2.0.2]   [spkg-install]       |     ptrdiff_t
  [numpy-2.0.2]   [spkg-install] /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/17/include/__stddef_ptrdiff_t.h:18:26: note: 'ptrdiff_t' declared here
  [numpy-2.0.2]   [spkg-install]    18 | typedef __PTRDIFF_TYPE__ ptrdiff_t;
  [numpy-2.0.2]   [spkg-install]       |                          ^
  [numpy-2.0.2]   [spkg-install] In file included from ../numpy/_core/src/umath/string_ufuncs.cpp:20:
  [numpy-2.0.2]   [spkg-install] ../numpy/_core/src/umath/string_fastsearch.h:219:23: error: invalid operands to binary expression ('CheckedIndexer<char>' and 'CheckedIndexer<char>')
  [numpy-2.0.2]   [spkg-install]   219 |             return (p - s);
  [numpy-2.0.2]   [spkg-install]       |                     ~ ^ ~
  [numpy-2.0.2]   [spkg-install] ../numpy/_core/src/umath/string_fastsearch.h:826:20: note: in instantiation of function template specialization 'rfindchar<char>' requested here
  [numpy-2.0.2]   [spkg-install]   826 |             return rfindchar(s_, n, p_[0]);
  [numpy-2.0.2]   [spkg-install]       |                    ^
  [numpy-2.0.2]   [spkg-install] ../numpy/_core/src/umath/string_buffer.h:898:19: note: in instantiation of function template specialization 'fastsearch<char>' requested here
  [numpy-2.0.2]   [spkg-install]   898 |             pos = fastsearch(start_loc, end_loc - start_loc, buf2.buf, buf2.after - buf2.buf, -1, FAST_SEARCH);
  [numpy-2.0.2]   [spkg-install]       |                   ^
  [numpy-2.0.2]   [spkg-install] ../numpy/_core/src/umath/string_ufuncs.cpp:1114:9: note: in instantiation of function template specialization 'string_find<ENCODING::ASCII>' requested here
  [numpy-2.0.2]   [spkg-install]  1114 |         string_find<ENCODING::ASCII>,
  [numpy-2.0.2]   [spkg-install]       |         ^
  [numpy-2.0.2]   [spkg-install] ../numpy/_core/src/umath/string_buffer.h:720:1: note: candidate template ignored: could not match 'Buffer' against 'CheckedIndexer'
  [numpy-2.0.2]   [spkg-install]   720 | operator-(Buffer<enc> lhs, Buffer<enc> rhs)
  [numpy-2.0.2]   [spkg-install]       | ^
  [numpy-2.0.2]   [spkg-install] ../numpy/_core/src/umath/string_buffer.h:736:1: note: candidate template ignored: could not match 'Buffer' against 'CheckedIndexer'
  [numpy-2.0.2]   [spkg-install]   736 | operator-(Buffer<enc> lhs, npy_int64 rhs)
  [numpy-2.0.2]   [spkg-install]       | ^
  [numpy-2.0.2]   [spkg-install] ../numpy/_core/src/umath/string_fastsearch.h:139:5: note: candidate function not viable: no known conversion from 'CheckedIndexer<char>' to 'size_t' (aka 'unsigned long') for 1st argument
  [numpy-2.0.2]   [spkg-install]   139 |     operator-(size_t rhs)
  [numpy-2.0.2]   [spkg-install]       |     ^         ~~~~~~~~~~
  [numpy-2.0.2]   [spkg-install] ../numpy/_core/src/umath/string_fastsearch.h:495:27: error: invalid operands to binary expression ('CheckedIndexer<char>' and 'CheckedIndexer<char>')
  [numpy-2.0.2]   [spkg-install]   495 |             return window - haystack;
  [numpy-2.0.2]   [spkg-install]       |                    ~~~~~~ ^ ~~~~~~~~
  [numpy-2.0.2]   [spkg-install] ../numpy/_core/src/umath/string_fastsearch.h:576:18: note: in instantiation of function template specialization '_two_way<char>' requested here
  [numpy-2.0.2]   [spkg-install]   576 |         result = _two_way(haystack + index,
  [numpy-2.0.2]   [spkg-install]       |                  ^
  [numpy-2.0.2]   [spkg-install] ../numpy/_core/src/umath/string_fastsearch.h:846:24: note: in instantiation of function template specialization '_two_way_count<char>' requested here
  [numpy-2.0.2]   [spkg-install]   846 |                 return _two_way_count(s_, n, p_, m, maxcount);
  [numpy-2.0.2]   [spkg-install]       |                        ^
  [numpy-2.0.2]   [spkg-install] ../numpy/_core/src/umath/string_buffer.h:898:19: note: in instantiation of function template specialization 'fastsearch<char>' requested here
  [numpy-2.0.2]   [spkg-install]   898 |             pos = fastsearch(start_loc, end_loc - start_loc, buf2.buf, buf2.after - buf2.buf, -1, FAST_SEARCH);
  [numpy-2.0.2]   [spkg-install]       |                   ^
  [numpy-2.0.2]   [spkg-install] ../numpy/_core/src/umath/string_ufuncs.cpp:1114:9: note: in instantiation of function template specialization 'string_find<ENCODING::ASCII>' requested here
  [numpy-2.0.2]   [spkg-install]  1114 |         string_find<ENCODING::ASCII>,
  [numpy-2.0.2]   [spkg-install]       |         ^
  [numpy-2.0.2]   [spkg-install] ../numpy/_core/src/umath/string_buffer.h:720:1: note: candidate template ignored: could not match 'Buffer' against 'CheckedIndexer'
  [numpy-2.0.2]   [spkg-install]   720 | operator-(Buffer<enc> lhs, Buffer<enc> rhs)
  [numpy-2.0.2]   [spkg-install]       | ^
  [numpy-2.0.2]   [spkg-install] ../numpy/_core/src/umath/string_buffer.h:736:1: note: candidate template ignored: could not match 'Buffer' against 'CheckedIndexer'
  [numpy-2.0.2]   [spkg-install]   736 | operator-(Buffer<enc> lhs, npy_int64 rhs)
  [numpy-2.0.2]   [spkg-install]       | ^
  [numpy-2.0.2]   [spkg-install] ../numpy/_core/src/umath/string_fastsearch.h:139:5: note: candidate function not viable: no known conversion from 'CheckedIndexer<char>' to 'size_t' (aka 'unsigned long') for 1st argument
  [numpy-2.0.2]   [spkg-install]   139 |     operator-(size_t rhs)
  [numpy-2.0.2]   [spkg-install]       |     ^         ~~~~~~~~~~
  [numpy-2.0.2]   [spkg-install] ../numpy/_core/src/umath/string_fastsearch.h:219:23: error: invalid operands to binary expression ('CheckedIndexer<unsigned int>' and 'CheckedIndexer<unsigned int>')
  [numpy-2.0.2]   [spkg-install]   219 |             return (p - s);
  [numpy-2.0.2]   [spkg-install]       |                     ~ ^ ~
  [numpy-2.0.2]   [spkg-install] ../numpy/_core/src/umath/string_fastsearch.h:826:20: note: in instantiation of function template specialization 'rfindchar<unsigned int>' requested here
  [numpy-2.0.2]   [spkg-install]   826 |             return rfindchar(s_, n, p_[0]);
  [numpy-2.0.2]   [spkg-install]       |                    ^
  [numpy-2.0.2]   [spkg-install] ../numpy/_core/src/umath/string_buffer.h:908:19: note: in instantiation of function template specialization 'fastsearch<unsigned int>' requested here
  [numpy-2.0.2]   [spkg-install]   908 |             pos = fastsearch((npy_ucs4 *)start_loc, end - start,
  [numpy-2.0.2]   [spkg-install]       |                   ^
  [numpy-2.0.2]   [spkg-install] ../numpy/_core/src/umath/string_ufuncs.cpp:1114:9: note: in instantiation of function template specialization 'string_find<ENCODING::ASCII>' requested here
  [numpy-2.0.2]   [spkg-install]  1114 |         string_find<ENCODING::ASCII>,
  [numpy-2.0.2]   [spkg-install]       |         ^
  [numpy-2.0.2]   [spkg-install] ../numpy/_core/src/umath/string_buffer.h:720:1: note: candidate template ignored: could not match 'Buffer' against 'CheckedIndexer'
  [numpy-2.0.2]   [spkg-install]   720 | operator-(Buffer<enc> lhs, Buffer<enc> rhs)
  [numpy-2.0.2]   [spkg-install]       | ^
  [numpy-2.0.2]   [spkg-install] ../numpy/_core/src/umath/string_buffer.h:736:1: note: candidate template ignored: could not match 'Buffer' against 'CheckedIndexer'
  [numpy-2.0.2]   [spkg-install]   736 | operator-(Buffer<enc> lhs, npy_int64 rhs)
  [numpy-2.0.2]   [spkg-install]       | ^
  [numpy-2.0.2]   [spkg-install] ../numpy/_core/src/umath/string_fastsearch.h:139:5: note: candidate function not viable: no known conversion from 'CheckedIndexer<unsigned int>' to 'size_t' (aka 'unsigned long') for 1st argument
  [numpy-2.0.2]   [spkg-install]   139 |     operator-(size_t rhs)
  [numpy-2.0.2]   [spkg-install]       |     ^         ~~~~~~~~~~
  [numpy-2.0.2]   [spkg-install] ../numpy/_core/src/umath/string_fastsearch.h:495:27: error: invalid operands to binary expression ('CheckedIndexer<unsigned int>' and 'CheckedIndexer<unsigned int>')
  [numpy-2.0.2]   [spkg-install]   495 |             return window - haystack;
  [numpy-2.0.2]   [spkg-install]       |                    ~~~~~~ ^ ~~~~~~~~
  [numpy-2.0.2]   [spkg-install] ../numpy/_core/src/umath/string_fastsearch.h:576:18: note: in instantiation of function template specialization '_two_way<unsigned int>' requested here
  [numpy-2.0.2]   [spkg-install]   576 |         result = _two_way(haystack + index,
  [numpy-2.0.2]   [spkg-install]       |                  ^
  [numpy-2.0.2]   [spkg-install] ../numpy/_core/src/umath/string_fastsearch.h:846:24: note: in instantiation of function template specialization '_two_way_count<unsigned int>' requested here
  [numpy-2.0.2]   [spkg-install]   846 |                 return _two_way_count(s_, n, p_, m, maxcount);
  [numpy-2.0.2]   [spkg-install]       |                        ^
  [numpy-2.0.2]   [spkg-install] ../numpy/_core/src/umath/string_buffer.h:908:19: note: in instantiation of function template specialization 'fastsearch<unsigned int>' requested here
  [numpy-2.0.2]   [spkg-install]   908 |             pos = fastsearch((npy_ucs4 *)start_loc, end - start,
  [numpy-2.0.2]   [spkg-install]       |                   ^
  [numpy-2.0.2]   [spkg-install] ../numpy/_core/src/umath/string_ufuncs.cpp:1114:9: note: in instantiation of function template specialization 'string_find<ENCODING::ASCII>' requested here
  [numpy-2.0.2]   [spkg-install]  1114 |         string_find<ENCODING::ASCII>,
  [numpy-2.0.2]   [spkg-install]       |         ^
  [numpy-2.0.2]   [spkg-install] ../numpy/_core/src/umath/string_buffer.h:720:1: note: candidate template ignored: could not match 'Buffer' against 'CheckedIndexer'
  [numpy-2.0.2]   [spkg-install]   720 | operator-(Buffer<enc> lhs, Buffer<enc> rhs)
  [numpy-2.0.2]   [spkg-install]       | ^
  [numpy-2.0.2]   [spkg-install] ../numpy/_core/src/umath/string_buffer.h:736:1: note: candidate template ignored: could not match 'Buffer' against 'CheckedIndexer'
  [numpy-2.0.2]   [spkg-install]   736 | operator-(Buffer<enc> lhs, npy_int64 rhs)
  [numpy-2.0.2]   [spkg-install]       | ^
  [numpy-2.0.2]   [spkg-install] ../numpy/_core/src/umath/string_fastsearch.h:139:5: note: candidate function not viable: no known conversion from 'CheckedIndexer<unsigned int>' to 'size_t' (aka 'unsigned long') for 1st argument
  [numpy-2.0.2]   [spkg-install]   139 |     operator-(size_t rhs)
  [numpy-2.0.2]   [spkg-install]       |     ^         ~~~~~~~~~~
  [numpy-2.0.2]   [spkg-install] In file included from ../numpy/_core/src/umath/string_ufuncs.cpp:21:
  [numpy-2.0.2]   [spkg-install] ../numpy/_core/src/umath/string_buffer.h:924:20: error: no matching function for call to 'string_find'
  [numpy-2.0.2]   [spkg-install]   924 |     npy_intp pos = string_find(buf1, buf2, start, end);
  [numpy-2.0.2]   [spkg-install]       |                    ^~~~~~~~~~~
  [numpy-2.0.2]   [spkg-install] ../numpy/_core/src/umath/string_ufuncs.cpp:1116:9: note: in instantiation of function template specialization 'string_index<ENCODING::ASCII>' requested here
  [numpy-2.0.2]   [spkg-install]  1116 |         string_index<ENCODING::ASCII>,
  [numpy-2.0.2]   [spkg-install]       |         ^
  [numpy-2.0.2]   [spkg-install] ../numpy/_core/src/umath/string_buffer.h:831:1: note: candidate template ignored: substitution failure [with enc = ENCODING::ASCII]
  [numpy-2.0.2]   [spkg-install]   831 | string_find(Buffer<enc> buf1, Buffer<enc> buf2, npy_int64 start, npy_int64 end)
  [numpy-2.0.2]   [spkg-install]       | ^
  [numpy-2.0.2]   [spkg-install] ../numpy/_core/src/umath/string_buffer.h:924:20: error: no matching function for call to 'string_find'
  [numpy-2.0.2]   [spkg-install]   924 |     npy_intp pos = string_find(buf1, buf2, start, end);
  [numpy-2.0.2]   [spkg-install]       |                    ^~~~~~~~~~~
  [numpy-2.0.2]   [spkg-install] ../numpy/_core/src/umath/string_ufuncs.cpp:1124:9: note: in instantiation of function template specialization 'string_index<ENCODING::UTF32>' requested here
  [numpy-2.0.2]   [spkg-install]  1124 |         string_index<ENCODING::UTF32>,
  [numpy-2.0.2]   [spkg-install]       |         ^
  [numpy-2.0.2]   [spkg-install] ../numpy/_core/src/umath/string_buffer.h:831:1: note: candidate template ignored: substitution failure [with enc = ENCODING::UTF32]
  [numpy-2.0.2]   [spkg-install]   831 | string_find(Buffer<enc> buf1, Buffer<enc> buf2, npy_int64 start, npy_int64 end)
  [numpy-2.0.2]   [spkg-install]       | ^
  [numpy-2.0.2]   [spkg-install] 7 errors generated.
```

--- a/build/pkgs/numpy/patches/27416.patch
+++ b/build/pkgs/numpy/patches/27416.patch
@@ -1,0 +1,22 @@
+From 84f53386bde9c994c49373075ce91353a7767bc8 Mon Sep 17 00:00:00 2001
+From: gorloffslava <31761951+gorloffslava@users.noreply.github.com>
+Date: Mon, 9 Sep 2024 12:37:49 +0500
+Subject: [PATCH] BUILD: fix missing include for std::ptrdiff_t for C++23
+ language mode
+
+---
+ numpy/_core/src/umath/string_fastsearch.h | 1 +
+ 1 file changed, 1 insertion(+)
+
+diff --git a/numpy/_core/src/umath/string_fastsearch.h b/numpy/_core/src/umath/string_fastsearch.h
+index 61abdcb5ad19..96c1e2d30140 100644
+--- a/numpy/_core/src/umath/string_fastsearch.h
++++ b/numpy/_core/src/umath/string_fastsearch.h
+@@ -9,6 +9,7 @@
+ #include <wchar.h>
+ 
+ #include <type_traits>
++#include <cstddef>
+ 
+ #include <numpy/npy_common.h>
+ 

